WinUIGallery/Helper/NavigationHelper.cs

```diff
@@ -215,14 +215,71 @@ private bool TryGoBack()
             return navigated;
         }
 
+        private bool TryGoForward()
+        {
+            bool navigated = false;
+            if (this.Frame.CanGoForward)
+            {
+                this.Frame.GoForward();
+                navigated = true;
+            }
+            return navigated;
+        }
+
         private void UpdateBackButton()
         {
-            if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 6))
+            this.CurrentNavView.IsBackEnabled = this.Frame.CanGoBack ? true : false;
+        }
+
+        /// <summary>
+        /// Invoked on every keystroke, including system keys such as Alt key combinations.
+        /// Used to detect keyboard navigation between pages even when the page itself
+        /// doesn't have focus.
+        /// </summary>
+        /// <param name="sender">Instance that triggered the event.</param>
+        /// <param name="e">Event data describing the conditions that led to the event.</param>
+
+        private void CurrentNavView_KeyDown(object sender, Microsoft.UI.Xaml.Input.KeyRoutedEventArgs e)
+        {
+            if (e.Handled || alreadyProcessedKeyDown)
             {
-               // this.CurrentNavView.IsBackButtonVisible = this.Frame.CanGoBack ? NavigationViewBackButtonVisible.Visible : NavigationViewBackButtonVisible.Collapsed;
-                this.CurrentNavView.IsBackEnabled = this.Frame.CanGoBack ? true : false;
+                return;
+            }
+
+            var virtualKey = e.Key;
+
+            // Only investigate further when Left, Right, or the dedicated Previous or Next keys
+            // are pressed
+            if (virtualKey == VirtualKey.Left || virtualKey == VirtualKey.Right ||
+                (int)virtualKey == 166 || (int)virtualKey == 167)
+            {
+                var downState = CoreVirtualKeyStates.Down;
+                bool menuKey = (InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Menu) & downState) == downState;
```

Comment:
The bools hear should follow the naming convention `isMenuKeyPreessed` or something like that.